### PR TITLE
cue.py: Support for gitlab

### PR DESCRIPTION
**Pull request by tboegi**
Add GITLAB_CI as another "ci context".

### CODE REVIEW

File: cue.py [hunk 1]
```diff
@@ -43,6 +43,12 @@ def detect_context():
             ci['compiler'] = os.environ['CMP']
         buildconfig = os.environ['CONFIGURATION'].lower()
 
+    if 'GITLAB_CI' in os.environ:
+        ci['service'] = 'gitlab'
+        ci['compiler'] = os.environ['CI_COMPILER']
```
Comment (ralphlange) [inline cue.py:48]: `CI_COMPILER` is not a predefined variable on GitLab. (see [GitLab documentation](https://gitlab.com/help/ci/variables/predefined_variables.md)).
In that case, I would prefer to use the variable `CMP` used on all other platforms. This keeps the `ci-scripts` documentation streamlined and short and helps users setting up things for multiple services.
```python
        if 'CMP' in os.environ:
            ci['compiler'] = os.environ['CMP']
```

File: cue.py [hunk 2]
```diff
@@ -43,6 +43,12 @@ def detect_context():
...
+        if 'BCFG' in os.environ:
+            buildconfig = os.environ['BCFG'].lower()
+
```
Comment (ralphlange) [inline cue.py:51]: Also needs settings for `ci['os']` and `ci['platform']`.

Comment (tboegi) [reply cue.py:51]: Similar question as above:
'os' seams  not to  be used within gitlab - should we hard code it to Linux ?
And what should we have in 'platform' ? Linux as well ?

Comment (ralphlange) [reply cue.py:51]: If Linux is what GitLab offers, hardcode `ci['os']` to `linux`, yes.
If GitLab only does 64bit, hardcode `ci['platform']` should be `x64`.

**Comment by ralphlange** [comment]
See: 0fc0dab7

**Comment by ralphlange** [comment]
Superseded by #55 - closing.

**Pull request closed without merging**